unistd/file.c

```diff
@@ -101,18 +101,26 @@ ssize_t __safe_read(int fd, void *buf, size_t size)
 			continue;
 		}
 		else {
-			break;
+			return -1;
```

Comment:
> Also fixed error handling in blocking safe read to always return -1 on error.

How is this different? The `else` branch is taken when `rlen < 0`, so `if (rlen < 0)` below is still executed.

Reply:
You're right  - disregard this comment.